lib/tasks/swagger_generate.rake

```diff
@@ -0,0 +1,276 @@
+      "responses" => {
+        201 => {
+          "description" => "#{klass_name} creation successful",
+          "schema" => {
+            "type" => "object",
+            "items" => {"$ref" => build_definition(klass_name)}
+          }
+        }
+      }
+    }
+  end
+
+  def swagger_update_description(klass_name, verb)
+    action = verb == "patch" ? "Update" : "Replace"
+    {
+      "summary" => "#{action} an existing #{klass_name}",
+      "operationId" => "#{action.downcase}#{klass_name}",
+      "description" => "#{action}s a #{klass_name} object",
+      "produces" => ["application/json"],
+      "consumes" => ["application/json"],
+      "parameters" => [
+        {"$ref" => build_parameter("ID")},
+        {
+          "name" => "body",
+          "in" => "body",
+          "description" => "#{klass_name} attributes to update",
+          "required" => true,
+          "schema" => {"$ref" => build_definition(klass_name)}
+        }
+      ],
+      "responses" => {
+        204 => {"description" => "Updated, no content"},
+        404 => {"description" => "Not found"}
+      }
+    }
+  end
+
+  def run
+    parameters["QueryOffset"] = {
+      "in" => "query",
+      "name" => "offset",
+      "type" => "integer",
+      "required" => false,
+      "default" => 0,
+      "minimum" => 0,
+      "description" => "The number of items to skip before starting to collect the result set."
+    }
+
+    parameters["QueryLimit"] = {
+      "in" => "query",
+      "name" => "limit",
+      "type" => "integer",
+      "required" => false,
+      "default" => 100,
+      "minimum" => 1,
+      "maximum" => 1000,
+      "description" => "The numbers of items to return per page."
+    }
+
+    definitions["CollectionLinks"] = {
+      "type" => "object",
+      "properties" => {
+        "first" => {
+          "type" => "string"
+        },
+        "last"  => {
+          "type" => "string"
+        },
+        "prev"  => {
+          "type" => "string"
+        },
+        "next"  => {
+          "type" => "string"
+        }
+      }
+    }
+
+    definitions["CollectionMetadata"] = {
+      "type" => "object",
+      "properties" => {
+        "count" => {
+          "type" => "integer"
+        }
+      }
+    }
+
+    definitions["OrderParameters"] = {
+      "type" => "object",
+      "properties" => {
+        "service_parameters" => {
+          "type" => "object",
+          "description" => "JSON object with provisioning parameters"
+        },
+        "provider_control_parameters" => {
+          "type" => "object",
+          "description" => "The provider specific parameters needed to provision this service. This might include namespaces, special keys"
+        }
+      }
+    }
+
+    new_content = swagger_contents
+    new_content["paths"] = build_paths.sort.to_h
+    new_content["parameters"] = parameters.sort.to_h.each_with_object({}) { |(name, val), h| h[name] = val || swagger_contents["parameters"][name] || {} }
```

Comment:
You don't need the .to_h after the sort, since you go straight into iterating over it anyway.